kotlin-react-router/src/jsMain/kotlin/react/router/IndexRouteObject.ext.kt

```diff
@@ -0,0 +1,14 @@
+@file:Suppress(
+    "CANNOT_CHECK_FOR_EXTERNAL_INTERFACE",
+)
+
+package react.router
+
+import kotlin.contracts.contract
+
+fun isIndexRouteObject(routeObject: RouteObject): Boolean {
```

Comment:
Looks like redundant checks

Reply:
Could you explain what is you use case, please?

Reply:
I can not just make single `RouteObject`, the interfaces are obviously different
![image](https://user-images.githubusercontent.com/15995890/219883877-e643c781-1ae8-4ca3-8b1a-816d83a0c8fd.png)

Reply:
I have no use cases for redundant guards